### PR TITLE
Organize third-party doc for better readability

### PR DESCRIPTION
**Pull request by raskasa**
Also:

* converted all entries to proper sentences.
* updated a few old links.

### CODE REVIEW

**Comment by raskasa** [comment]
1. Should we remove `metrics-influxdb` now that there is a first-class module available?
2. There are currently two histogram entries - which one should be deleted?
3. Are `camel-metrics` and `MetricsCatcher` under the correct sections?
3. In general, is this organization good enough?  (i.e. better section names wanted, entries under the incorrect section, etc.)

File: docs/source/manual/third-party.rst [hunk 1]
```diff
@@ -7,36 +7,47 @@ Third Party Libraries
...
-* `metrics-spring <https://github.com/ryantenney/metrics-spring>`_ provides integration with Spring
-* `sematext-metrics-reporter <https://github.com/sematext/sematext-metrics-reporter>`_ provides a reporter for `SPM <http://sematext.com/spm/index.html>`_.
-* `wicket-metrics <https://github.com/NitorCreations/wicket-metrics>`_ provides easy integration for your `Wicket <http://wicket.apache.org/>`_ application.
+Instrumented Libraries
+~~~~~~~~~~~~~~~~~~~~~~
+
+* `camel-metrics <https://github.com/InitiumIo/camel-metrics>`_ provides component for your `Apache Camel <https://camel.apache.org/>`_ route.
+* `hdrhistogram-metrics-reservoir <https://bitbucket.org/marshallpierce/hdrhistogram-metrics-reservoir>`_ provides a Metrics reservoir implementation backed by `HdrHistogram <http://hdrhistogram.org/>`_.
+* `hdrhistogram-metrics-reservoir <https://bitbucket.org/marshallpierce/hdrhistogram-metrics-reservoir>`_ provides a Histogram reservoir backed by `HdrHistogram <http://hdrhistogram.org/>`_.
```
Comment (marshallpierce) [inline docs/source/manual/third-party.rst]: this one has a slightly more useful description than the one on line 14, I think, so let's keep this one and ditch that one

Comment (raskasa) [inline docs/source/manual/third-party.rst]: Fixed.

**Comment by ryantenney** [comment]
Influxdb isn't a first party module until v4, so we'll keep it in here for now.
camel-metrics is in the right place, MetricsCatcher isn't, but it would be in a category all its own. It's fine there for now
This organization is already far better than what we had, and we can make further improvements in the future.  Thanks for doing this!

**Pull request merged**